src/util/ImageUploader.cpp

```diff
@@ -0,0 +1,118 @@
+#include "util/ImageUploader.hpp"
+
+#include "singletons/Settings.hpp"
+
+#include <QJsonDocument>
+#include <QJsonObject>
+
+namespace chatterino {
+namespace imageuploader {
+namespace detail {
+
+QJsonObject exportSettings(Settings &s)
+{
+    QJsonObject settingsObj;
+    settingsObj["Version"] = "1.0.0";
+    settingsObj["Name"] = "Chatterino Image Uploader Settings";
+    settingsObj["RequestMethod"] = "POST";
+    settingsObj["RequestURL"] = s.imageUploaderUrl.getValue();
+    settingsObj["Body"] = "MultipartFormData";
+    settingsObj["FileFormName"] = s.imageUploaderFormField.getValue();
+    settingsObj["URL"] = s.imageUploaderLink.getValue();
+    settingsObj["DeletionURL"] = s.imageUploaderDeletionLink.getValue();
+
+    QString headers = s.imageUploaderHeaders.getValue();
+    if (!headers.isEmpty()) {
+        QJsonObject headersObj;
+        QStringList headerLines = headers.split('\n', Qt::SkipEmptyParts);
+        for (const QString &line : headerLines) {
+            QStringList parts = line.split(':', Qt::SkipEmptyParts);
+            if (parts.size() >= 2) {
+                QString key = parts[0].trimmed();
+                QString value = parts.mid(1).join(':').trimmed();
+                headersObj[key] = value;
+            }
+        }
+        if (!headersObj.isEmpty()) {
+            settingsObj["Headers"] = headersObj;
+        }
+    }
+
+    return settingsObj;
+}
+
+bool importSettings(const QJsonObject &settingsObj, Settings &s)
+{
+    bool hasValidSettings = false;
```

Comment:
Are there any "meta" fields that _always_ need to be part of the export that we can check?
e.g. if the json blob always needs `Name` and `Version`, we could check those here and error out early if something is wrong

Reply:
Only the meta fields from ShareX exports, which I think are `Version` and `Name`